### PR TITLE
build(gradle): add license plugin

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] documentation is changed or added

##### Description of change

Adds gradle license plugin, the Apache 2 short license notice, and formats all files in the project.

:notebook: N.B. Java files have been switched from JavaDoc style to SlashStar style to prevent conflicts between the license plugin and the static analysis plugin.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,6 +15,7 @@ plugins {
     id 'com.github.sherter.google-java-format' version '0.6' apply false
     id 'nebula.lint' version '8.3.1' apply false
     id 'net.ltgt.errorprone' version '0.0.11' apply false
+    id 'com.github.hierynomus.license' version '0.14.0' apply false
```
Comment (apetro) [inline build.gradle]: <Insert standard grumpiness about production-grade products (uPortal) depending on pre-release `0.whatever` versioned stuff.>

Comment (ChristianMurphy) [reply build.gradle]: :man_shrugging: the [other plugin](https://github.com/jaredsburrows/gradle-license-plugin) I found is still at `0.6.0` with only 1 contributor. 
If you have a more stable plugin that could be used, suggestions are welcome :smile:

Comment (ChristianMurphy) [reply build.gradle]: Its also worth noting that https://github.com/hierynomus/license-gradle-plugin
is a wrapper around the classes from http://code.mycila.com/license-maven-plugin, which is at a very stable 3.0

Comment (apetro) [reply build.gradle]: Yup. I do wish projects would heed [SemVer advice about when to 1.0.0](http://semver.org/#how-do-i-know-when-to-release-100).

**Comment by ChristianMurphy** [comment]
There are merge conflicts and build errors blocking this PR.
This out, will open a revised version as time permits.

**Pull request closed without merging**